src/main/java/com/rabbitmq/http/client/ReactorNettyClient.java

```diff
@@ -511,98 +503,101 @@ public static String basicAuthentication(String username, String password) {
     }
 
     private <T> Mono<T> doGetMono(Class<T> type, String... pathSegments) {
-        return client.get(uri(pathSegments), request -> Mono.just(request)
-            .transform(this::addAuthorization)
-            .flatMap(HttpClientRequest::send))
-            .onErrorMap(this.errorHandler)
-            .transform(decode(type));
+        return client.headers(authorizedHeader())
+            .doOnResponse(onResponseCallback)
```

Comment:
I've noticed an error we are trying to fix before 0.8 where the callback is not invoked.

Reply:
Should we wait for a fix or is there an alternative API we can use?

Reply:
Tests are passing, so it looks like it's working in this case. The `doOnResponse` calls some logic that can deal with 4xx and 5xx responses.